fd/diff.go

```diff
@@ -45,8 +49,7 @@ type Settings struct {
 // of the function.
 func DefaultSettings() *Settings {
```

Comment:
With the checks done at the beginning of `Derivative` and `Gradient`, this function is actually not necessary because zero value of `Settings` is acceptable as a default.